via/services/google_drive.py

```diff
@@ -0,0 +1,123 @@
+import json
+import re
+from io import BytesIO
+from json import JSONDecodeError
+from typing import ByteString, Iterator
+
+from google.oauth2.service_account import Credentials
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaIoBaseDownload
+from pyramid.httpexceptions import HTTPNotFound
+
+from via.exceptions import ConfigurationError, UpstreamServiceError
+
+
+class GoogleDriveAPI:
+    """Simplified interface for interacting with Google Drive."""
+
+    SCOPES = [
+        # If we want metadata
+        "https://www.googleapis.com/auth/drive.metadata.readonly",
+        # To actually get the file
+        "https://www.googleapis.com/auth/drive.readonly",
+    ]
+
+    def __init__(self, service_account_info):
+        """Initialise the service.
+
+        :param service_account_info: A dict of credentials info as provided by
+            Google console's JSON format.
+
+        :raises ConfigurationError: If the credentials are not accepted by Google
+        """
+        if service_account_info:
+            try:
+                credentials = Credentials.from_service_account_info(
+                    service_account_info, scopes=self.SCOPES
+                )
+            except ValueError as exc:
+                raise ConfigurationError(
+                    "The Google Drive service account information is invalid"
+                ) from exc
+
+            self._service = build("drive", "v3", credentials=credentials)
+        else:
+            self._service = None
+
+    @property
+    def is_available(self):
+        """Get whether the Google Drive API is available."""
+
+        return self._service is not None
+
+    _PUBLIC_URL_REGEX = re.compile(
+        r"^https://drive.google.com/uc\?id=(.*)&export=download$", re.IGNORECASE
+    )
```

Comment:
I think the use of a regex here comes from [existing code](https://github.com/hypothesis/via/blob/f6383f688034d3af536407d5bba45983f4604a26/via/get_url/details.py#L19-L21) but just FYI [we need to replace this with proper URL parsing](https://github.com/hypothesis/via/issues/569). Can be a separate issue

Reply:
I'm working on a separate PR for this. I could have just smooshed them together honestly... but never mind. It's not a change actually related to this PR, so it's probably good form not to.